shell/mixins/resource-fetch.js

```diff
@@ -92,35 +100,53 @@ export default {
           this.hasManualRefresh = true;
         }
       }
+    },
+    // data fetching for the mechanism
+    $fetchType(type, multipleResources = [], storeType) {
+      const currStore = storeType || this.$store.getters['currentStore']();
+
+      this.$initializeFetchData(type, multipleResources, currStore);
```

Comment:
If we call this each type we request a primary or secondary resource, won't the `hasManualRefresh` and `incremental` flags be relevant to whatever was last called? 

Could we get around this by setting those flags to true if any primary or secondary resource is over the watch/incremental count...
```
this.watch = this.watch || watch;
```
kind of thing?

That could lead us to the situation where the primary is tiny but secondary/s are large... (shell/list/node.vue primary node, secondary pods)... and we enable the optimisations.

The alternative is if we restrict the watch/increment to only primary, which would have the reverse affect (node.vue... primary node is small so won't enable optimisations... that large pod secondary won't have)

Reply:
> If we call this each type we request a primary or secondary resource, won't the hasManualRefresh and incremental flags be relevant to whatever was last called?

No because the `this.$initializeFetchData` runs only once (under an init flag) and that's what sets the `incremental`, `manual`, `watch`, etc.

> Could we get around this by setting those flags to true if any primary or secondary resource is over the watch/incremental count...
>
> this.watch = this.watch || watch;
>
> kind of thing?
> 
> That could lead us to the situation where the primary is tiny but secondary/s are large... (shell/list/node.vue primary node, secondary pods)... and we enable the optimisations.
> 
> The alternative is if we restrict the watch/increment to only primary, which would have the reverse affect (node.vue... primary node is small so won't enable optimisations... that large pod secondary won't have)

The manual refresh flag is bound to the primary resource, so that cannot be changed to subsequent resources, imo, following this issue briefing.

As to larger datasets of secondary data like `PODS`, we can have enabled the incremental loading so that it loads and updates the DOM more frequently. I am just not sure what it "hurts" more in terms of performance: 1 large non-UI-blocking request or multiple DOM updates.

If you believe that this issue can be moved forward, what do you think about creating a subsequent issue where we analise this further?

Reply:
RE $initializeFetchData :man_facepalming: good point. sorry, reviewed in a hurry